docs/Contributing-back.rst

```diff
@@ -0,0 +1,496 @@
+~~~~~~~~~~~~~~
+
+.. _contributing_back_code:
+
+Code contributions
+~~~~~~~~~~~~~~~~~~
+
+.. _contributing_back_easyconfigs:
+
+Easyconfig contributions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _contributing_back_docs:
+
+Documentation
+~~~~~~~~~~~~~
+
+
+.. _contributing_back_pull_requests:
+
+Pull requests
+-------------
+
+To contribute back to the EasyBuild framework, easyblocks, easyconfig files
+or the EasyBuild documentation, you will need to open a pull request to the
+corresponding GitHub repository:
+
+* EasyBuild framework: https://github.com/hpcugent/easybuild-framework
+* easyblocks: https://github.com/hpcugent/easybuild-easyblocks
+* easyconfigs: https://github.com/hpcugent/easybuild-easyconfigs
+* documentation: https://github.com/hpcugent/easybuild (see ``docs`` subdirectory)
+
+Doing this the traditional way requires some knowledge about using ``git``
+(i.e. creating commits, using branches, pushing to and pulling from remote
+Git repositories, etc.), and being familiar with GitHub.
+
+However, this can be largely circumvented by using the GitHub integration
+provided by EasyBuild, see :ref:`integration_with_github`.
+
+
+.. _contributing_back_pull_request_setup:
+
+Setting up
+~~~~~~~~~~
+
+.. note:: These instructions assume that you already have a working GitHub account.
+
+          If you don't have a GitHub account yet, visit https://github.com/ to create one.
+
+          Make sure you have registered your SSH public key in your GitHub account,
+          via https://github.com/settings/keys.
+
+Before you can open pull requests, you will need to **fork** the GitHub repository
+you are targeting, and create a local **working copy** of that repository. This only
+needs to be done *once* for every of the EasyBuild repositories.
+
+.. note:: For the purpose of this guide, we will be using the ``easybuild-framework`` repository.
+
+          However, the instructions are analogous for the other EasyBuild repositories.
+
+.. _contributing_back_pull_request_setup_fork_repo:
+
+Forking the repository
+++++++++++++++++++++++
+
+First, create your own *fork* of the repository in your GitHub account using the
+'``Fork``' button on the top right at https://github.com/hpcugent/easybuild-framework.
+
+This will basically create a 'copy' of the ``easybuild-framework`` repository
+owned by the ``hpcugent`` GitHub organisation to your own personal GitHub account.
+
+You will have to use this fork as a staging area for your work,
+to prepare your contributions before creating the actual pull requests.
+
+.. _contributing_back_pull_request_setup_working_copy:
+
+Creating a working copy
++++++++++++++++++++++++
+
+In addition to forking the repository on GitHub, you also need to create
+a local *working copy* of the repository. This is basically a local checkout
+of the repository where you can track the changes you are making.
+
+First, *clone* the repository:
+
+.. code:: shell
+
+  git clone git@github.com:hpcugent/easybuild-framework.git
```

Comment:
For using the manual approach, I find this confusing too. If you clone your own fork, you don't care about the original repo and don't have to mess with remotes (at least until your copy is out of sync). Wouldn't it make more sense to keep it to the basics now, and simply follow the structure below? It is already complicated enough without introducing additional remotes

fork -> clone own fork -> check out develop -> create new branch -> modify -> commit -> push -> make PR using the webinterface